File: (03-18-2024)Two Sum/joshlin/solution.ts
function twoSum(nums: number[], target: number): number[] {
    // time: O(n), space: O(n)
    let complement: {[key: number]: number} = {}
    for (const [index, num] of nums.entries()) {
        const diff = target - num
        if (diff in complement) {
            return [index, complement[diff]]
        } else {
            complement[num] = index
        }
    }
    return []
};

File: (03-18-2024)Two Sum/joshlin/solution.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # time: O(n), space: O(n)
        complement = {}
        for index, num in enumerate(nums):
            diff = target - num
            if diff in complement:
                return [index, complement[diff]]
            else:
                complement[num] = index